src/goto-instrument/synthesizer/loop_invariant_synthesizer.cpp

```diff
@@ -0,0 +1,46 @@
+/*******************************************************************\
+
+Module: Synthesize Loop Invariants
+
+Author: Qinheping Hu
+
+\*******************************************************************/
+
+/// \file
+/// Synthesize Loop Invariants
+
+#include "loop_invariant_synthesizer.h"
+
+void enumerative_loop_invariant_synthesizert::init_candidates()
+{
+  for(auto &function_p : goto_model.goto_functions.function_map)
+  {
+    natural_loops_mutablet natural_loops(function_p.second.body);
```

Comment:
I don't think you really want the `mutable` variant here?

Reply:
Yes, Thank you!